.github/workflows/static-checks.yaml

```diff
@@ -0,0 +1,44 @@
+name: static-checks
+on:
+  pull_request:
+  push:
+    branches:
+      - 'release-*'
+      - 'master'
+      - 'main'
+    tags:
+      - 'v*'
+env:
+  golang-version: '1.16'
+jobs:
+  golang-lint:
+    runs-on: ubuntu-latest
+    name: Golang linter
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.42.1
+          args: --timeout 10m0s
+      - name: Help
```

Comment:
Not sure if there is a better way to provide hints to the author on how to fix the problem.

Reply:
IIRC the GitHub action will also add comments to the pull request?